pkg/agent/runner.go

```diff
@@ -59,6 +59,21 @@ func (a *agentSpecRunner) RunTask(ctx context.Context, prompt string) (AgentResu
 		}
 	}
 
+	// Create an empty temporary directory for agent execution to isolate it from source code
+	tempDir, err := os.MkdirTemp("", "gevals-agent-")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create temporary directory for agent execution: %w", err)
+	}
+	executionSucceeded := false
+	defer func() {
+		// Clean up temp directory unless GEVALS_DEBUG is set and execution failed
+		// In that case, preserve it for debugging
+		shouldPreserve := os.Getenv("GEVALS_DEBUG") != "" && !executionSucceeded
+		if !shouldPreserve {
+			_ = os.RemoveAll(tempDir)
+		}
```

Comment:
I think this will be hard to use in debugging, as we don't surface to users the name of the temp dir that was created in the case of an error (as far as I can tell).

IMO we should either:
1. Always clean up the temp dir after agent execution
2. Surface the name of the temp dir to the runner somehow so that it can be shown to the user

Reply:
Maybe a simple approach is to append some text saying where the agent dir was created to the output string? That way, if someone is looking at the agent output trying to debug, the relevant info will be there